compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -900,6 +902,95 @@ public BLangNode transform(ObjectTypeDescriptorNode objTypeDescNode) {
         return deSugarTypeAsUserDefType(objectTypeNode);
     }
 
+    public BLangObjectTypeNode createObjectExpressionBody(NodeList<Node> members) {
+        BLangObjectTypeNode objectTypeNode = (BLangObjectTypeNode) TreeBuilder.createObjectTypeNode();
+        objectTypeNode.flagSet.add(Flag.ANONYMOUS);
+
+        for (Node node : members) {
+            BLangNode bLangNode = node.apply(this);
+            NodeKind nodeKind =  bLangNode.getKind();
+            if (nodeKind == NodeKind.FUNCTION) {
+                BLangFunction bLangFunction = (BLangFunction) bLangNode;
+                bLangFunction.attachedFunction = true;
+                bLangFunction.flagSet.add(Flag.ATTACHED);
+                if (!Names.USER_DEFINED_INIT_SUFFIX.value.equals(bLangFunction.name.value)) {
+                    objectTypeNode.addFunction(bLangFunction);
+                    continue;
+                }
+                if (objectTypeNode.initFunction != null) {
+                    objectTypeNode.addFunction(bLangFunction);
+                    continue;
+                }
+                if (bLangFunction.requiredParams.size() != 0) {
+                    dlog.error(bLangFunction.pos, DiagnosticCode.OBJECT_CTOR_INIT_CANNOT_HAVE_PARAMETERS);
+                    continue;
+                }
+                bLangFunction.objInitFunction = true;
+                objectTypeNode.initFunction = bLangFunction;
+            } else if (nodeKind == NodeKind.VARIABLE) {
+                objectTypeNode.addField((BLangSimpleVariable) bLangNode);
+            } else if (nodeKind == NodeKind.USER_DEFINED_TYPE) {
+                dlog.error(bLangNode.pos, DiagnosticCode.OBJECT_CTOR_DOES_NOT_SUPPORT_TYPE_REFERENCE_MEMBERS);
+            }
+        }
+
+        objectTypeNode.isAnonymous = true;
+        return objectTypeNode;
+    }
+
+    @Override
+    public BLangNode transform(ObjectConstructorExpressionNode objectConstructorExpressionNode) {
+
+        DiagnosticPos pos = getPositionWithoutMetadata(objectConstructorExpressionNode);
+
+        BLangObjectTypeNode objectTypeNode = createObjectExpressionBody(objectConstructorExpressionNode.members());
+        objectTypeNode.pos = pos;
+        BLangObjectConstructorExpression objectCtorExpression = TreeBuilder.createObjectCtorExpression(objectTypeNode);
```

Comment:
Any other method in TreeBuilder doesn't seem to take arguments. I'm not completely sure why it was designed like that but I think it would be better to adhere to that constraint.

Reply:
We don't seem to use this `objectCtorExpression` again, shall we remove this maybe